Puzzlepart/did
# Project Role Management Enhancements

### Your checklist for this pull request
- [x] Make sure strings/resources are added using our [[resource files](https://github.com/Puzzlepart/did/tree/dev/.resources)](https://github.com/Puzzlepart/did/tree/dev/.resources)
- [x] Make sure [[CHANGELOG.md](https://github.com/Puzzlepart/did/blob/dev/CHANGELOG.md)](https://github.com/Puzzlepart/did/blob/dev/CHANGELOG.md) is updated if applicable
- [x] Make sure [[Smoke tests](https://github.com/Puzzlepart/did/blob/dev/.github/ISSUE_TEMPLATE/smoke_test.md)](https://github.com/Puzzlepart/did/blob/dev/.github/ISSUE_TEMPLATE/smoke_test.md) are updated if applicable
 
### Review checklist

### Review theme song
🎵 [Imagine Dragons - Believer](https://open.spotify.com/track/0pqnGHJpmpxLKifKRmU6P6) 🎵

### Description
This pull request introduces a series of enhancements to the project role management functionality. These changes streamline the process of assigning and modifying roles, improve the user experience through more intuitive UI elements, and strengthen underlying logic to ensure roles reflect project requirements accurately.  

### How to test
1. Check out this branch locally using `gh pr checkout 1247`.
2. Run `npm run watch` to compile and monitor changes.
3. Navigate to the project role management interface.
4. Test assigning, removing, and updating roles across a range of project scenarios.
5. Verify that UI labels and instructions are clear, that changes are accurately logged, and that no unexpected errors occur.

### Related issues
Closes #1247

### Related discussions
_None_

### Related PRs
_None_.
